inclusive-design/wecount.inclusivedesign.ca
# feat: added pagination component

## Description

Add pagination component. This component should be used on individual posts and pages which list results, such as "News and Views" and search/tag pages.

## Criteria

- [x] Previous and next links must link to the previous/next post or page of results.
- [x] Previous and next link text must be consistent.
- [x] Previous link must only appear if there is one or more prior post or page of results.
- [x] Next link must only appear if there is one or more subsequent post or page of results.
- [x] If the current item is the first item in a sequence, the `current` link should be the first numeric page link.
- [x] If the current item is the last item in a sequence, the `current` link should be the last numeric page link.
- [x] Numeric page links must include `aria-label` attributes with the correct page number.
- [x] Ellipses should only appear in numeric page links if there are more than three pages of results before or after to the current item.
- [x] On "News and Views", search, and tag pages, the NewsGrid should be broken into chunks of 6 items and paginated accordingly.

## Steps to test

1. Visit a single news item.
2. Visit "News and Views" page.
3. Visit search results page for `inclusion`.
4. Visit tag page for `AI`.

**Expected behavior:** Paginated navigation should work according to the criteria above in all four contexts.

## Additional information

Not applicable.

## Related issues

Resolves #65.
